### PR TITLE
docs/nia: new configuration for services condition & source_input

### PR DESCRIPTION
**Pull request by lornasong**
Add configuration docs for new fields for `condition "services"` and `source_input "services"`
 - Datacenter
 - Namespace
 - Filter
 - CTS User Defined Meta

Changes:
 - Update `condition "services"` config to use new table format
 - Add docs for new configuration field - reuses language from other configs for Datacenter and Namespace
 - Update `service` block's parallel `filter` and `cts_user_defined_meta` with similar config language (keep old-style list format)

Preview links: [`service` block](https://consul-qlvy9vaj1-hashicorp.vercel.app/docs/nia/configuration#filter), [`condition "services"`](https://consul-qlvy9vaj1-hashicorp.vercel.app/docs/nia/configuration#services-condition), and [`source_input "services"`](https://consul-qlvy9vaj1-hashicorp.vercel.app/docs/nia/configuration#services-source-input)

### CODE REVIEW

**Review by findkim — APPROVED** [review]

**Comment by lornasong** [comment]
Rebased to latest

**Comment by lornasong** [comment]
@findkim apologies, I realized I should have included the corresponding configuration changes for source_input. Would you be able to re-review when you have a chance? I added an additional commit https://github.com/hashicorp/consul/pull/11646/commits/0846b73a03fb3ae91304e500f4811aad4b2f63e6 for source_input which has the same content as condition that you already looked at. Here's the [source_input preview link](https://consul-51v6nl6d0-hashicorp.vercel.app/docs/nia/configuration#services-source-input)

**Review by findkim — APPROVED** [review]

**Comment by lornasong** [comment]
Hello Consul Education team! I know y'all are pretty busy. I wanted to double-check if anyone thought they would give any feedback on this documentation PR?

I'm happy to hold-off merging if there are plans to review. If not, that's totally okay. I'm planning to merge this next week Tuesday morning if I don't hear that I should hold off. Thanks!

**Comment by trujillo-adam** [comment]
@lornasong - Looking now

File: website/content/docs/nia/configuration.mdx [hunk 1]
```diff
@@ -314,7 +326,11 @@ This `services` source input object defines services registered to Consul whose
...
+| `regexp` |  Optional | String value matching the names of Consul services to monitor. Only services that have a name matching the regular expression are used by the task. <br/><br/> If `regexp` is configured, then [`task.services`](#services) list must be omitted or empty. <br/><br/> If both a list and a regex are needed, consider including the list as part of the regex or creating separate tasks. | none |
+| `datacenter` |  Optional | String value specifying the name of a datacenter to query for the task. | Datacenter of the agent that Consul-Terraform-Sync queries. |
+| `namespace`  | Optional | <EnterpriseAlert inline /> <br/> String value indicating the namespace of the services to query for the task. | In order of precedence: <br/> 1. Inferred from the Consul-Terraform-Sync ACL token <br/> 2. The `default` namespace. |
+| `filter` |  Optional | String value specifying the expression used to filter nodes for the service. <br/><br/> For more details, see the Consul documentation on [supported filters for services](/api-docs/health#filtering-2) and [how to write filter expressions](/api-docs/features/filtering#creating-expressions) | none |
```
Comment (trujillo-adam) [inline website/content/docs/nia/configuration.mdx:332]: Same comment about consistency here

**Review by trujillo-adam — APPROVED** [review]
Added some suggestions

File: website/content/docs/nia/configuration.mdx [hunk 2]
```diff
@@ -118,8 +118,8 @@ service {
...
 - `namespace` <EnterpriseAlert inline /> - (string: "default") The namespace of the service. If not provided, the namespace will be inferred from the Consul-Terraform-Sync ACL token, or default to the `default` namespace.
-- `filter` - (string) Specifies the expression used to filter nodes for the service. For more details on supported filters, see the Consul documentation on [filtering service nodes](/api-docs/health#filtering-2).
-- `cts_user_defined_meta` - (map[string]) User-defined metadata is a map of strings that will be appended to the [service input variable](/docs/nia/installation/requirements#module-specifications) for compatible Terraform modules. Not all modules may use this value. To determine if your task uses metadata or what the expected keys and format are, reference documentation for the module(s) configured for your tasks.
+- `filter` - (string) Specifies the expression used to filter nodes for the service. For more details, see the Consul documentation on [supported filters for services](/api-docs/health#filtering-2) and [how to write filter expressions](/api-docs/features/filtering#creating-expressions).
```
[the remark this answers is not in the document]
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx]: Thanks for the suggestion! I'm realizing I wasn't clear about `filter`'s relationship with the other fields. `filter` provides additional ways to filter the services that are monitored e.g. `datacenter` filters the services by datacenter. I also realized I didn't include the field type (string).

I added to your suggestion as such:

> String value specifying an expression used to additionally filter which services to monitor. Refer to the [services filtering documentation](/api-docs/health#filtering-2) and section about [how to write filter expressions](/api-docs/features/filtering#creating-expressions) for additional information.

Feedback welcome!

File: website/content/docs/nia/configuration.mdx [hunk 3]
```diff
@@ -118,8 +118,8 @@ service {
...
+- `cts_user_defined_meta` - (map[string]) User-defined metadata is a map of strings that will be appended to the [service input variable](/docs/nia/terraform-modules#services-source-input) for compatible Terraform modules. Not all modules use the configured metadata. Refer to the module configured for this task to understand the metadata usage and expected keys and format.
```
[the remark this answers is not in the document]
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx]: Thanks! This is a lot clearer.

File: website/content/docs/nia/configuration.mdx [hunk 4]
```diff
@@ -206,12 +206,24 @@ task {
...
+    filter     = "Service.Tags not contains \"prod\""
+    cts_user_defined_meta {
+      key = "value"
+    }
   }
 }
 ```
 
-- `regexp` - **(beta)** (string) Only services that have a name which matches the regular expression are used by the task. If `regexp` is configured, then [`task.services`](#services) must be omitted or empty. If both a list and a regex are needed, consider including the list as part of the regex or creating separate tasks.
+| Parameter |  Required |Description | Default |
```
[the remark this answers is not in the document]
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:221]: Glad to hear! I actually borrowed this table format from elsewhere in our docs so can't claim the credit for it :)

**Pull request merged**